Hypothesis Testing assignment with Buyer Ratio.csv.
H0 = There is no difference between male female buyer ratio.
H1 = There is a significant difference.

We are doing an Chi square the data is catagorical.
And comparing the p value with alpha.

In [5]:
import pandas as pd
import numpy as np
import scipy
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency
data=pd.read_csv('BuyerRatio.csv')
data.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [6]:
# Make dimensional array
obs=np.array([[50,142,131,70],[435,1523,1356,750]])
obs

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [7]:
# Chi2 contengency independence test
chi2_contingency(obs) # o/p is (Chi2 stats value, p_value, df, expected obsvations)

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

We found that the p value of 0.66 is greater than 0.05. So we cannot reject the null hypothesis.
It proves that there is no significant difference between male female buyer ratio.